stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel_actions/ChannelActionsDialogFragment.kt

```diff
@@ -0,0 +1,146 @@
+        setupViews()
+        channelActionsViewModel.members.observe(viewLifecycleOwner) {
+            membersAdapter.submitList(it)
+            bindMemberNames(it)
+            bindMembersInfo(it)
+        }
+    }
+
+    override fun getTheme(): Int = R.style.StreamChannelActionsBottomSheetDialog
+
+    private fun setupViews() {
+        if (isGroup) {
+            binding.leaveGroupButton.isVisible = true
+            binding.deleteChannelButton.isVisible = true
+            binding.deleteContactButton.isVisible = false
+            binding.leaveGroupButton.setOnClickListener {
+                listener?.onLeaveGroupClicked(cid)
+            }
+            binding.deleteChannelButton.setOnClickListener {
+                listener?.onDeleteChannelClicked(cid)
+            }
+        } else {
+            binding.leaveGroupButton.isVisible = false
+            binding.deleteChannelButton.isVisible = false
+            binding.deleteContactButton.isVisible = true
+            binding.deleteContactButton.setOnClickListener {
+                listener?.onDeleteContactClicked(cid)
+            }
+        }
+        binding.recyclerView.adapter = membersAdapter
+    }
+
+    private fun bindMemberNames(members: List<Member>) {
+        binding.channelMembersTextView.text = if (isGroup) {
+            members.joinToString { it.user.name }
+        } else {
+            members.getOrNull(0)?.user?.name ?: ""
+        }
+    }
+
+    private fun bindMembersInfo(members: List<Member>) {
+        binding.membersInfoTextView.text = if (isGroup) {
+            requireContext().resources.getQuantityString(
+                R.plurals.stream_channel_actions_members_count,
+                members.size,
+                members.size,
+                members.count { it.user.online }
+            )
+        } else {
+            members
+                .getOrNull(0)
+                ?.user
+                ?.lastActive
+                ?.time
+                ?.let { DateUtils.getRelativeTimeSpanString(it) }
+                ?.let { requireContext().getString(R.string.stream_channel_actions_last_seen, it) }
+                ?: ""
+        }
+    }
+
+    companion object {
+        private const val ARG_CID = "cid"
+        private const val ARG_IS_GROUP = "is_group"
+
+        fun newGroupChatInstance(cid: String): ChannelActionsDialogFragment {
+            return newInstance(cid, true)
+        }
+
+        fun newDirectMessageInstance(cid: String): ChannelActionsDialogFragment {
+            return newInstance(cid, false)
+        }
+
+        private fun newInstance(cid: String, isGroup: Boolean): ChannelActionsDialogFragment {
+            return ChannelActionsDialogFragment().apply {
+                arguments = Bundle().apply {
+                    putString(ARG_CID, cid)
+                    putBoolean(ARG_IS_GROUP, isGroup)
+                }
+            }
+        }
+    }
+}
+
+public interface OnChannelActionSelectedListener {
```

Comment:
Why is this `public` if it's only being used in the `Fragment` above, and that's `internal`? Also, if it belongs to that Fragment, shouldn't we nest it inside it?

Reply:
Done